docs/ai/azure-ai-services-authentication.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Authenticate to Azure OpenAI using .NET
+description: Learn about the different options to authenticate to Azure OpenAI and other services using .NET
+author: alexwolfmsft
+ms.topic: conceptual
+ms.date: 06/27/2024
+---
+
+# Azure AI services authentication and authorization using .NET
+
+Application requests to Azure AI Services must be authenticated. In this article, you explore the options available to authenticate to Azure OpenAI and other AI services using .NET. These concepts apply to the Semantic Kernel SDK, as well as SDKs from specific services such as Azure OpenAI. Most AI services offer two primary ways to authenticate apps and users:
+
+- **Key-based authentication** provides access to an Azure service using secret key values. These secret values are sometimes known as API keys or access keys depending on the service.
+- **Microsoft Entra ID** provides a comprehensive identity and access management solution to ensure that the correct identities have the correct level of access to different Azure resources.
+
+The sections ahead provide conceptual overviews for these two approaches, rather than detailed implementation steps. For more detailed information about connecting to Azure services, visit the following resources:
+
+- [Authenticate .NET apps to Azure services](/dotnet/azure/sdk/authentication/)
+- [Identity fundamentals](/entra/fundamentals/identity-fundamental-concepts)
+- [What is Azure RBAC?](/azure/role-based-access-control/overview)
+
+> [!NOTE]
+> The examples in this article focus primarily on connections to Azure OpenAI, but the same concepts and implementation steps directly apply to many other Azure AI Services as well.
+
+## Authentication using keys
+
+Access keys allow apps and tools to authenticate to an Azure AI service such as OpenAI using a secret key provided by the service. Retrieve the secret key using tools such as the Azure Portal or Azure CLI and use it to configure your app code to connect to the AI service:
+
+```csharp
+builder.Services.AddAzureOpenAIChatCompletion(
+    "deployment-model",
+    "service-endpoint",
+    "service-key"); // Secret key
+var kernel = builder.Build();
+```
+
+Using keys is a straightforward option, but this approach should be used with caution. Keys aren't the recommended authentication option for the following reasons:
+
+- Keys don't follow [the principle of least privilege](/entra/identity-platform/secure-least-privileged-access)—they provide elevated permissions regardless of who uses them or for what task.
+- Keys can accidentally be checked into source control or stored in unsafe locations.
+- Keys can easily be shared with or sent to parties who shouldn't have access.
+- Keys often require manual administration and rotation.
```

Comment:
```suggestion
Using keys is a straightforward option, but this approach should be used with caution. Keys aren't the recommended authentication option because they:

- Don't follow [the principle of least privilege](/entra/identity-platform/secure-least-privileged-access)—they provide elevated permissions regardless of who uses them or for what task.
- Can accidentally be checked into source control or stored in unsafe locations.
- Can easily be shared with or sent to parties who shouldn't have access.
- Often require manual administration and rotation.
```